src/docs/development/tools/devtools/code-size.md

```diff
@@ -0,0 +1,194 @@
+---
+title: Using the Code Size Tool
+description: Learn how to use the DevTools code size tool.
+---
+
+## What is it?
+
+The code size tool allows you to visualize, analyze, and diff code / app size
+data.
+
+## How to use it
+
+If DevTools is already connected to a running application, navigate to the
+"Code Size" tab.
+
+![Screenshot of code size tab]({% asset tools/devtools/code_size_tab.png @path %})
+
+If DevTools is not connected to a running application, you can
+access the tool from the landing page that appears once you have launched
+DevTools.
+
+![Screenshot of code size access on landing page]({% asset tools/devtools/code_size_access_landing_page.png @path %})
+
+## Analysis Tab
+
+![Screenshot of code size analysis]({% asset tools/devtools/code_size_analysis.png @path %}){:width="100%"}
+
+### Loading a code size file
+
+When you open the Analysis tab, you'll see instructions to load a code size
+file. Drag and drop a code size file in, and click "Analyze Size".
+
+![Screenshot of code size analysis loading screen]({% asset tools/devtools/code_size_load_analysis.png @path %}){:width="100%"}
+
+See [Generating size files][] below for information on generating size files.
+
+### Treemap map and table
+The treemap and table show the hierarchical data for your app's size.
+
+For questions about using the treemap, see [How to use the treemap][] below.
+
+### Dominator Tree and Call Graph
+
+This section of the page shows code size attribution data. This data is visible
+in the form of a dominator tree as well as a call graph.
+
+#### Using the dominator tree
+
+![Screenshot of code size dominator tree]({% asset tools/devtools/code_size_dominator_tree.png @path %}){:width="100%"}
+
+A dominator tree is a tree where each node's children are those nodes it
+immediately dominates. A node `d` dominates a node `n` if every entry point to
+`n` must go through `d`.
+
+To put it in context of app size analysis, imagine `package:a` imports both
+`package:b` and `package:c`, and both `package:b` and `package:c` import
+`package:d`.
+
+```
+package:a
+|__ package:b
+|   |__ package:d
+|__ package:c
+    |__ package:d
+```
+
+In this example, `package:a` dominates `package:d`, so the dominator tree for
+this data would look like:
+
+```
+package:a
+|__ package:b
+|__ package:c
+|__ package:d
+```
+
+This information is helpful for understanding why certain pieces of code are
+present into your compiled application. For example, if you are analyzing your
+app size and find an unexpected package included in your compiled app, you can
+use the dominator tree to trace the package to its root source.
+
+#### Using the call graph
+
+![Screenshot of code size call graph]({% asset tools/devtools/code_size_call_graph.png @path %}){:width="100%"}
+
```

Comment:
Add a few words to describe that it's an alternate view to the dominator tree used for the same purpose of determining why something is in the code.

Reply:
added info - ptal